src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.cs

```diff
@@ -21,7 +30,7 @@ namespace Microsoft.CodeAnalysis
     /// This type generally assumes that files on disk aren't changing, since it ensure that two calls to <see cref="LoadFromPath(string)"/>
     /// will always return the same thing, per that interface's contract.
     /// </remarks>
-    internal abstract partial class AnalyzerAssemblyLoader : IAnalyzerAssemblyLoader
+    internal abstract partial class AnalyzerAssemblyLoader : IAnalyzerAssemblyLoaderInternal
```

Comment:
I was a little surprised I didn't see any changes which type-test or cast to `IAnalyzerAssemblyLoaderInternal`. I thought we had public API which permits users to pass in instances of `IAnalyzerAssemblyLoader` in order to use a common analyzer loader across compilations.

Reply:
You're correct that we have end points that provide `IAnalyzerAssemblyLoader` parameters. This particular API is only used in the compiler server where we do not provide a public API end point. It's not a re-hostable component but one where we provide all the implementations. 

Pretty much the only reason that the compiler server takes `IAnalyzerAssemblyLoader` and not `AnalyzerAssemblyLoader` is that it limits the API surface area and it's easier to provide throwing implementations in unit tests so we can validate behavior better.